src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -35,6 +36,13 @@ internal class BuildEventArgsWriter
         /// </summary>
         private readonly MemoryStream currentRecordStream;
 
+        /// <summary>
+        /// For NameValueList we need to prefix the storage size
+        ///  (distinct from values count due to variable int encoding)
+        /// So using same technique as with 'currentRecordStream'.
+        /// </summary>
+        private readonly MemoryStream nameValueListStream = new MemoryStream(256);
```

Comment:
nit: Initialize the field in the constructor for consistency with `currentRecordStream`? Magic numbers look better close to each other 😉